jlm/mlir/backend/mlirgen.hpp

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2023 Magnus Sjalander <work@sjalander.com>
+ * See COPYING for terms of redistribution.
+ */
+
+#ifndef JLM_MLIR_BACKEND_MLIRGEN_HPP
+#define JLM_MLIR_BACKEND_MLIRGEN_HPP
+
+//#include <jlm/llvm/ir/operators/GetElementPtr.hpp>
+//#include <jlm/llvm/ir/operators/load.hpp>
+//#include <jlm/llvm/ir/operators/operators.hpp>
+//#include <jlm/llvm/ir/operators/sext.hpp>
+//#include <jlm/llvm/ir/operators/store.hpp>
+//#include <jlm/rvsdg/bitstring/comparison.hpp>
+//#include <jlm/rvsdg/bitstring/type.hpp>
+
+#include <jlm/llvm/ir/operators/lambda.hpp>
+#include <jlm/llvm/ir/RvsdgModule.hpp>
+
+#ifdef MLIR
+
+#include <RVSDG/RVSDGDialect.h>
+#include <RVSDG/RVSDGPasses.h>
+#include <JLM/JLMDialect.h>
+
+#include "mlir/Dialect/Arith/IR/Arith.h"
+
+//#include "mlir/IR/Builders.h"
+//#include "mlir/IR/BuiltinTypes.h"
+//#include "mlir/IR/BuiltinOps.h"
+//#include "mlir/IR/Dialect.h"
+//#include "mlir/IR/MLIRContext.h"
+//#include "mlir/InitAllDialects.h"
+//#include "mlir/InitAllPasses.h"
+//#include "mlir/Pass/Pass.h"
+//#include "mlir/Pass/PassManager.h"
+//#include "mlir/Support/FileUtilities.h"
+//#include "mlir/Tools/mlir-opt/MlirOptMain.h"
+//#include "llvm/Support/CommandLine.h"
+//#include "llvm/Support/InitLLVM.h"
+//#include "llvm/Support/SourceMgr.h"
+//#include "llvm/Support/ToolOutputFile.h"
+//#include "mlir/Tools/mlir-opt/MlirOptMain.h"
+
+namespace jlm {
+namespace rvsdgmlir {
+
+  class MLIRGen
+  {
+  public:
+    MLIRGen() {
+      mlir::MLIRContext * context = new mlir::MLIRContext();
+      // Load the RVSDG dialect
+      context->getOrLoadDialect<mlir::rvsdg::RVSDGDialect>();
+      // Load the JLM dialect
+      context->getOrLoadDialect<mlir::jlm::JLMDialect>();
+      // Load the Arith dialect
+      context->getOrLoadDialect<mlir::arith::ArithDialect>();
+      Builder_ = std::make_unique<mlir::OpBuilder>(context);
```

Comment:
Is the Builder the owner of the context after you hand it over? In other words, does it take care of the deallocation of the context? If so, then this is okay. If not, then you have a memory leak again.